### PR TITLE
user-util: add new wrappers for reading/writing {passwd,shadow,gshado…

### PR DESCRIPTION
**Pull request by fbuihuu**
…w} database files

The API povided by the glibc is too error-prone as one has to deal directly
with errno in order to detect if errors occured.

### CODE REVIEW

File: src/basic/user-util.c [hunk 1]
```diff
@@ -734,3 +734,125 @@ bool synthesize_nobody(void) {
...
+}
+#endif
+
+int fgetpwent_sane(FILE *stream, struct passwd **pw) {
+        struct passwd *p;
+
+        assert(pw);
+        assert(stream);
+
+        if ((p = fgetpwent(stream)) == NULL) {
```
Comment (keszybz) [inline src/basic/user-util.c]: I'd very much prefer to have the assignment in a separate line, and the check as `!p`.

File: src/basic/user-util.c [hunk 2]
```diff
@@ -734,3 +734,125 @@ bool synthesize_nobody(void) {
...
+                if (errno == ENOENT)
+                        return false;
+                if (errno == 0)
```
Comment (keszybz) [inline src/basic/user-util.c]: Please use the same form as above: `return errno > 0 ? -errno :  -EIO`.
(This help the compiler know that the return value is guaranteed to be negative.)

Here, and below.

File: src/basic/user-util.h [hunk 3]
```diff
@@ -24,6 +24,10 @@
 #include <stdint.h>
 #include <sys/types.h>
 #include <unistd.h>
+#include <grp.h>
+#include <gshadow.h>
+#include <pwd.h>
+#include <shadow.h>
```
Comment (keszybz) [inline src/basic/user-util.h]: Those should be sorted alphabetically with the rest...

**Comment by fbuihuu** [comment]
@keszybz, thanks for the feedback.

Done.

**Pull request merged**